### PR TITLE
feat(test-runner): add coverageConfig.reporters configuration option

### PR DESCRIPTION
**Pull request by MathieuPuech**
## What I did

1. add a reporters option for coverageConfig allowing to use [alternative istanbul reporters](https://istanbul.js.org/docs/advanced/alternative-reporters/) for code coverage

### CODE REVIEW

File: .changeset/rude-countries-pretend.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@web/test-runner': minor
+'@web/test-runner-core': minor
```
Comment (LarsDenBakker) [inline .changeset/rude-countries-pretend.md:3]: this should be a patch since we're still > 0.x.y

**Comment by LarsDenBakker** [comment]
Looks good, thanks! Is it possible to add reporters other than the ones built-in from istanbul?

**Comment by MathieuPuech** [comment]
I don’t find any documentation about that. I’ve found this custom reporter:
https://github.com/Elgolfin/istanbul-reporter-shield-badge#readme

It seems to be possible to register a reporter like that, I can add some code to enable this in the PR

**Comment by MathieuPuech** [comment]
this package is a wrong example since it's based on the deprecated [istanbul](https://www.npmjs.com/package/istanbul) package.

A package using the good version could be coveradge but it seems to need some configuration when created:

https://github.com/brettz9/coveradge/blob/master/src/reporter.js

I will check about theses missing options

**Comment by MathieuPuech** [comment]
it seems that istanbul reporters could need to options when using the `create` function.

See https://github.com/istanbuljs/nyc/blob/91ae8b8764cef361fe6add31ae0802dc221fb577/index.js#L457

I've added the options needed for `coveradge`

You can use `coveradge` installing in dev dependencies and add in the `reporters` array of `coverageConfig` the string `coveradge`.

The coveradge package ignore the context and generate the badge in the project root but it seems to be an issue on this package.

**Comment by dwilson6** [comment]
Thanks for implementing this! I was just looking to see if passing custom reporters was possible.

**Pull request merged**